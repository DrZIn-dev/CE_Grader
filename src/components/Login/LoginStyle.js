import styled, { createGlobalStyle } from "styled-components";
import { Window, WindowHeader, reset, Button } from "react95";
const ResetStyles = createGlobalStyle`
  ${reset}
`;

const WindowStyled = styled(Window)`
  width: 400px;
`;

const WindowHeaderStyled = styled(WindowHeader)`
  display: flex;
  align-items: center;
  justify-content: space-between;
`;

const ButtonStyled = styled(Button)`
  margin-right: -6px;
  margin-top: 1px;
`;

const Spanstyled = styled.span`
  font-weight: bold;
  transform: translateY(-1px);
`;

const Div = styled.div`
  position: fixed;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
`;

export {
  WindowStyled,
  Div,
  ButtonStyled,
  WindowHeaderStyled,
  ResetStyles,
  Spanstyled
};
