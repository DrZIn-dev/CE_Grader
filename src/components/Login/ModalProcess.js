import React from 'react'
import styled, { ThemeProvider } from 'styled-components'

import { Progress, themes, WindowContent } from 'react95'

import {
    Div,
    ResetStyles,
    WindowHeaderStyled,
    WindowStyled,
} from './LoginStyle'

const WindowContentStyled = styled(WindowContent)`
	display: flex;
	flex-direction: row;
	justify-content: space-between;
	align-items: center;
`

const Modal = ({ progress }) => {
    return (
        <Div style={{ zIndex: 3 }}>
            <ResetStyles />
            <ThemeProvider theme={themes.default}>
                <WindowStyled>
                    <WindowHeaderStyled>
                        <span>
                            In process.
						</span>
                    </WindowHeaderStyled>
                    <WindowContentStyled>
                            <Progress percent={progress} shadow width={350} />
                    </WindowContentStyled>
                </WindowStyled>
            </ThemeProvider>
        </Div>
    )
}

export default Modal