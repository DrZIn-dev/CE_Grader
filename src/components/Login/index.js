import React from "react";

import styled from "styled-components";

const Div = styled.div`
  background-color: teal;
`;

const ExtraDiv = styled(Div)`
  background-color: red;
  font-size: 30px;
`;

export default () => {
  return (
    <div>
      This is div
      <Div>This is Div with styled</Div>
      <ExtraDiv>This is Extra Div styled</ExtraDiv>
    </div>
  );
};
