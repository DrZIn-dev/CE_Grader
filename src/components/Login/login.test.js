import React from "react";
import { shallow, mount, render } from "enzyme";

import Login from "./index";

describe("Login", () => {
  it("should render ", () => {
    const component = shallow(<Login />);
    expect(component).toMatchSnapshot();
  });

  it("should mount", () => {
    const component = mount(<Login />);
    expect(component).toMatchSnapshot();
  });

  it("should render", () => {
    const component = render(<Login />);
    expect(component).toMatchSnapshot();
  });
});
