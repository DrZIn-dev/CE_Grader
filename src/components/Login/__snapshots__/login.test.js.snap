// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login should mount 1`] = `ReactWrapper {}`;

exports[`Login should render  1`] = `ShallowWrapper {}`;

exports[`Login should render 1`] = `
initialize {
  "0": Object {
    "attribs": Object {},
    "children": Array [
      Object {
        "data": "This is div",
        "next": Object {
          "attribs": Object {
            "class": "sc-bdVaJa jtOXgR",
          },
          "children": Array [
            Object {
              "data": "This is Div with styled",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "text",
            },
          ],
          "name": "div",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": Object {
            "attribs": Object {
              "class": "sc-bdVaJa sc-bwzfXH ugvnr",
            },
            "children": Array [
              Object {
                "data": "This is Extra Div styled",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "text",
              },
            ],
            "name": "div",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": [Circular],
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
            },
          },
          "parent": [Circular],
          "prev": [Circular],
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
          },
        },
        "parent": [Circular],
        "prev": null,
        "type": "text",
      },
      Object {
        "attribs": Object {
          "class": "sc-bdVaJa jtOXgR",
        },
        "children": Array [
          Object {
            "data": "This is Div with styled",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "text",
          },
        ],
        "name": "div",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": Object {
          "attribs": Object {
            "class": "sc-bdVaJa sc-bwzfXH ugvnr",
          },
          "children": Array [
            Object {
              "data": "This is Extra Div styled",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "text",
            },
          ],
          "name": "div",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": null,
          "parent": [Circular],
          "prev": [Circular],
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
          },
        },
        "parent": [Circular],
        "prev": Object {
          "data": "This is div",
          "next": [Circular],
          "parent": [Circular],
          "prev": null,
          "type": "text",
        },
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
        },
      },
      Object {
        "attribs": Object {
          "class": "sc-bdVaJa sc-bwzfXH ugvnr",
        },
        "children": Array [
          Object {
            "data": "This is Extra Div styled",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "text",
          },
        ],
        "name": "div",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": Object {
          "attribs": Object {
            "class": "sc-bdVaJa jtOXgR",
          },
          "children": Array [
            Object {
              "data": "This is Div with styled",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "text",
            },
          ],
          "name": "div",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": [Circular],
          "parent": [Circular],
          "prev": Object {
            "data": "This is div",
            "next": [Circular],
            "parent": [Circular],
            "prev": null,
            "type": "text",
          },
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
          },
        },
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
        },
      },
    ],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {},
    "x-attribsPrefix": Object {},
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
